atlas/AtlasMl/atlasml/clients/artemis_client.py

```
@@ -0,0 +1,212 @@
+
+
+class CompetencyMapping(BaseModel):
+    competency_id: str
+    exercise_id: int
+    course_id: str
+
+
+class ArtemisAPIClient:
+    """Client for interacting with Artemis API for course and exercise operations."""
+
+    def __init__(self, base_url: str = None, api_token: str = None):
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)